### ISSUE TITLE
lovefield.min.js does not support aliases for aggregate fields...

### ISSUE THREAD

**Issue by johndimm** [google/lovefield#255]
... but lovefield.js does.

The problem can be seen in the quick start todo demo.  Replace the last statement with this:

```js
  return db.insertOrReplace().into(item).values([row]).exec();
}).then(function() {
  return todoDb.select(
    lf.fn.count().as('count of rows')
  ).from(item).where(item.done.eq(false)).exec();
}).then(function(results) {
  results.forEach(function(row) {
    console.log(row['count of rows']);
  });
```

That works with lovefield.js (console log shows the correct count of 1), but fails on lovefield.min.js:

   lovefield.min.js:28 Uncaught TypeError: lf.fn.count(...).as is not a function

**Comment by freshp86** [comment]
Thanks for the report!

Given that @export annotation is used at https://github.com/google/lovefield/blob/6e6cc7cae1421a8da6d5e55ffecb0cdedf6b29a6/lib/aggregated_column.js#L101, Closure compiler should preserve `as()` when minify-ing, but apparently this is not the case. Need to investigate further, and potentially build a minimal repro against the compiler.

**Comment by freshp86** [comment]
I am not able to reproduce the problem. Seems to work fine for me with lovefield.min.js. See complete example below.

```js
var schemaBuilder = lf.schema.create('todo', 1);

schemaBuilder.createTable('Item').
    addColumn('id', lf.Type.INTEGER).
    addColumn('description', lf.Type.STRING).
    addColumn('deadline', lf.Type.DATE_TIME).
    addColumn('done', lf.Type.BOOLEAN).
    addPrimaryKey(['id']).
    addIndex('idxDeadline', ['deadline'], false, lf.Order.DESC);

var todoDb;
var item;

schemaBuilder.connect().then(function(db) {
  todoDb = db;
  item = db.getSchema().table('Item');
  var row = item.createRow({
    'id': 1,
    'description': 'Get a cup of coffee',
    'deadline': new Date(),
    'done': false
  });

  return db.insertOrReplace().into(item).values([row]).exec();
}).then(function() {
  return todoDb.select(
    lf.fn.count(item.id).as('count of rows')
  ).from(item).where(item.done.eq(false)).exec();
}).then(function(results) {
  console.log(JSON.stringify(results));
});
```

Which prints the following in the devtools console
```
[{"count of rows":1}]
```

Which version of Lovefield are you using? I tried both the master branch, as well the 2.1.12 and both work fine.

**Comment by johndimm** [comment]
You are correct!  The version that fails is the one in the codelab stock_analyzer.  The current minified version works.   
(My bad.  Thought I had tested the current one but the old one must have been cached.)

Bad version:  

-rw-r-----@ 1 johndimm  staff  116374 May 26  2015 lovefield.min.js

$ wc -l lovefield.min.js

     345 lovefield.min.js

Good version:

-rw-r--r--  1 johndimm  staff  142677 Jul 17 10:28 lovefield.min.js.new

$ wc -l lovefield.min.js.new

     276 lovefield.min.js.new

  


    On Friday, July 19, 2019, 12:40:02 PM PDT, freshp86 <notifications@github.com> wrote:  
 
 
I am not able to reproduce the problem. Seems to work fine for me with lovefield.min.js. See complete example below.
var schemaBuilder = lf.schema.create('todo', 1);

schemaBuilder.createTable('Item').
    addColumn('id', lf.Type.INTEGER).
    addColumn('description', lf.Type.STRING).
    addColumn('deadline', lf.Type.DATE_TIME).
    addColumn('done', lf.Type.BOOLEAN).
    addPrimaryKey(['id']).
    addIndex('idxDeadline', ['deadline'], false, lf.Order.DESC);

var todoDb;
var item;

schemaBuilder.connect().then(function(db) {
  todoDb = db;
  item = db.getSchema().table('Item');
  var row = item.createRow({
    'id': 1,
    'description': 'Get a cup of coffee',
    'deadline': new Date(),
    'done': false
  });

  return db.insertOrReplace().into(item).values([row]).exec();
}).then(function() {
  return todoDb.select(
    lf.fn.count(item.id).as('count of rows')
  ).from(item).where(item.done.eq(false)).exec();
}).then(function(results) {
  console.log(JSON.stringify(results));
});
Which prints the following in the devtools console
[{"count of rows":1}]

Which version of Lovefield are you using? I tried both the master branch, as well the 2.1.12 and both work fine.
